### PR TITLE
ci: yet more enhancements / chore: more fixed lints

### PR DESCRIPTION
**Pull request by neersighted**
Also contains minor reformats/refactors.

* Improve flake8 config (almost fully commented/de-culted)
* Add flake8-comprehensions and fix lints.
* Add flake8-typing-imports and fix lints.
* Strip meaningless noqa using yesqa.
* Remove blanket noqas and replace type comments with annotations
* Have Github Actions annotate test failures.
* Drop the no-op Code Quality workflow.
* Add some additional flake8 lints (no implicit namespaces, no backslashes, no untidy imports (e.g. no `import foo.bar as bar`).

I've touched install-poetry.py -- once I've got the lints in this repo to satisfaction I plan to PR them to install.python-poetry.org with the install-poetry.py changes.

### CODE REVIEW

File: tests/installation/test_installer.py [hunk 1]
```diff
@@ -593,15 +593,14 @@ def test_run_install_with_optional_group_selected(
...
-            [tuple()],
+    list(
+        itertools.chain(
+            [()],
             itertools.permutations(RESERVED_PACKAGES, 1),
             itertools.permutations(RESERVED_PACKAGES, 2),
             [RESERVED_PACKAGES],
         )
-    ],
+    ),
```
Comment (danieleades) [inline tests/installation/test_installer.py:603]: can this method accept the iterator directly?

```python
    itertools.chain(
        [()],
        itertools.permutations(RESERVED_PACKAGES, 1),
        itertools.permutations(RESERVED_PACKAGES, 2),
        [RESERVED_PACKAGES],
    )
```

File: tests/installation/test_installer_old.py [hunk 2]
```diff
@@ -295,15 +295,14 @@ def test_run_install_no_group(installer, locker, repo, package, installed):
...
-            [tuple()],
+    list(
+        itertools.chain(
+            [()],
             itertools.permutations(RESERVED_PACKAGES, 1),
             itertools.permutations(RESERVED_PACKAGES, 2),
             [RESERVED_PACKAGES],
         )
-    ],
+    ),
```
Comment (danieleades) [inline tests/installation/test_installer_old.py:305]: as above. do you need the `list`, or does the naked iterator do the job?

File: .flake8 [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
     E501,
-    # W503: Line break occurred before a binary operator
-    W503
+    # E203: Whitespace before ':' (pycqa/pycodestyle#373)
+    E203,
 per-file-ignores =
-    # F401: Module imported by unused
+    # F401: Module imported by unused (non-implicit modules)
     __init__.py:F401
 exclude =
```
[the remark this answers is not in the document]
Comment (neersighted) [reply .flake8:12]: Aha! Beauty!

File: .flake8 [hunk 4]
```diff
@@ -1,21 +1,18 @@
...
-    # E501: Line too long
+max-line-length = 88
+inline-quotes = double
+extend-ignore =
+    # E501: Line too long (FIXME: long string constants)
     E501,
-    # W503: Line break occurred before a binary operator
-    W503
+    # E203: Whitespace before ':' (pycqa/pycodestyle#373)
+    E203,
```
Comment (abn) [inline .flake8:9]: any reason why we are adding this ignore now?

File: .flake8 [hunk 5]
```diff
@@ -1,21 +1,18 @@
...
-    .mypy_cache,
-    .pytest_cache,
-    .tox,
-    .venv,
-    .vscode,
-    __pycache__,
-    dist,
+    # INP001: File is part of an implicit namespace package.
+    get-poetry.py:INP001,
+    install-poetry.py:INP001,
```
Comment (abn) [inline .flake8:15]: I would exclude these files altogether as these should be really frozen now.

File: .flake8 (hunk 4 above)
Comment (neersighted) [reply .flake8:9]: It's part of the standard black config. We haven't triggered it, but black calls for it as it's a known conflict.

File: .flake8 (hunk 5 above)
Comment (neersighted) [reply .flake8:15]: Sounds good.

File: .flake8 (hunk 4 above)
Comment (neersighted) [reply .flake8:9]: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8

**Review by danieleades — APPROVED** [review]
LGTM

**Review by abn — APPROVED** [review]

**Pull request merged**